cli/src/cmd_net.rs

```diff
@@ -1507,3 +1864,72 @@ async fn current_port_settings(
     let settings = create_current(id, client).await?;
     Ok(settings)
 }
+
+/// Link forward error correction setting.
+#[derive(clap::ValueEnum, Clone, Copy, Debug)]
+pub enum CliLinkFec {
+    /// Firecode forward error correction.
+    Firecode,
+    /// No forward error correction.
+    None,
+    /// Reed-Solomon forward error correction.
+    Rs,
+}
+
+impl From<CliLinkFec> for LinkFec {
+    fn from(value: CliLinkFec) -> Self {
+        match value {
+            CliLinkFec::Firecode => LinkFec::Firecode,
+            CliLinkFec::None => LinkFec::None,
+            CliLinkFec::Rs => LinkFec::Rs,
+        }
+    }
+}
```

Comment:
FWIW, https://github.com/oxidecomputer/oxide.rs/blob/main/sdk/src/clap_feature.rs show another way to address what I infer to be the problem you were solving here, but I'm not sure.